src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpProxyConnectionHandler.cs

```diff
@@ -62,19 +64,35 @@ private Task<HttpResponseMessage> GetConnectionAndSendAsync(HttpRequestMessage r
         private async Task<HttpResponseMessage> SendWithProxyAsync(
             Uri proxyUri, HttpRequestMessage request, CancellationToken cancellationToken)
         {
+            HttpResponseMessage response;
+            HttpRequestMessage savedRequest = null;
+            HttpConnectionPool sslPool = null;
+
             if (proxyUri.Scheme != UriScheme.Http)
             {
                 throw new InvalidOperationException(SR.net_http_invalid_proxy_scheme);
             }
 
-            if (!HttpUtilities.IsSupportedNonSecureScheme(request.RequestUri.Scheme))
+            if (HttpUtilities.IsSupportedNonSecureScheme(request.RequestUri.Scheme))
             {
-                // TODO #23136: Implement SSL tunneling through proxy
-                throw new NotImplementedException("no support for SSL tunneling through proxy");
+                response = await GetConnectionAndSendAsync(request, proxyUri, cancellationToken).ConfigureAwait(false);
+            } else {
+                string sslHostName = ConnectHelper.GetSslHostName(request);
```

Comment:
I still think this seems like the wrong place for this logic.

This seems like it should be encapsulated down in the CreateConnectionAsync code in HttpConnectionPool.